### ISSUE TITLE
Preservica sync testing

### ISSUE THREAD

**Issue by mikeapp** [yalelibrary/YUL-DC#2039]
**Story**

We need to coordinate time to test against the actual Preservica instance instead of fixtures.   This will require modification of asset(s) in Preservica.

**Acceptance**

Test creating new parents with Preservica source
- [x] Create new parent
- [x] Create parent with large number of children (> 500 images)
- [x] Create parent with large (size) child objects

Test resync scenarios:
- [x] Test workflows for this work in Preservica
- [x] No change rsync
- [x] Resync against a existing DCS parent that is not already connected with Preservica, to make sure this sync attempt fails
- [x] Add children rsync (waiting on #2083)
- [x] Delete children rsync (waiting on #2083)
- Resync when parent in Preservica needs to be fully deleted and reingested (see new ticket #2137)
- Resync when child object is replaced in Preservica (not possible as replace function in Preservica is broken)
- [x] Resync when children are reordered in Preservica

**Comment by motropuk** [comment]
Testing plan https://docs.google.com/document/d/1r_vRj1j01PYdphDWdZu0kBJuHeg5KudRU4uC1OzPWoU/edit. Need to discuss with DPS to ask which Preservica workflows we should be using for some of these use cases in V6, e.g replacing a child, or adding a child etc. Workflows for this are different in V6 so need to ask DPS for advice for moving forward

**Comment by motropuk** [comment]
Mark mentioned Ebony magazine as option for large ingest testing

**Comment by motropuk** [comment]
Still need to do some testing in Preservica, but some highlights from our meeting with David yesterday
- It is not possible to add or delete children in Structure 2 in Preservica. The only option for adding files to this structure is to delete the entire object and reingest everything again. It sounds like Structure 2 will not be used for many Preservica packages, and we should advise people adding content to DCS to package new content in Preservica using structure 1
- There is a workflow in V6 for replacing child objects with new versions. However, this is not working currently. This is a known issue Preservica are working on fixing in V6. Until this fix is in place, again stakeholders would need to delete the parent object in Preservica and reingest
- It is not possible to rename files in Preservica for structure 2, to change sort order. So again reingest would be the only option in this case for this structure.
- More people need to be able to approve deletions in Preservica. See #2060

**Comment by motropuk** [comment]
Update: Working on create parent for Structure 1. Getting an error "Empty list of attributes passed". @K8Sewell is looking into this

**Comment by motropuk** [comment]
Created #2072

**Comment by motropuk** [comment]
Created #2074

**Comment by motropuk** [comment]
Created #2078

**Comment by motropuk** [comment]
Created #2079

**Comment by motropuk** [comment]
Created #2083

**Comment by motropuk** [comment]
Martin created #2090

**Comment by motropuk** [comment]
Same issue. Sync creates parent for added child in DCS, but does not add image and batch process seems to be hanging https://collections-uat.library.yale.edu/management/batch_processes/1076/parent_objects/900140256

I just dont have enough info to know what exactly is happening and where the hold up is - on the Preservica side or DCS side. Kait to explore

**Comment by motropuk** [comment]
waiting on #2109 and #2105

**Comment by motropuk** [comment]
Discussions with DPS staff indicates that best approach when wanting to make changes to an object in Preservica, is to reingest the whole object and then delete the old object. So this changes the emphasis of our testing a little and I have reordered what we will test when

**Comment by motropuk** [comment]
Started a documentation document for the DCS Preservica integration https://docs.google.com/document/d/1_Hx8YQYO56EK_1-rQnBTSKQ5m_khhjY4ACMKeXW8oeU/edit?usp=sharing

**Comment by motropuk** [comment]
Started a documentation document for the DCS Preservica integration https://docs.google.com/document/d/1_Hx8YQYO56EK_1-rQnBTSKQ5m_khhjY4ACMKeXW8oeU/edit?usp=sharing

**Comment by motropuk** [comment]
Started a documentation document for the DCS Preservica integration https://docs.google.com/document/d/1_Hx8YQYO56EK_1-rQnBTSKQ5m_khhjY4ACMKeXW8oeU/edit?usp=sharing

**Comment by motropuk** [comment]
Alicia created a parent with over 500 children https://collections-uat.library.yale.edu/catalog/900144552 so checking that test off

**Comment by motropuk** [comment]
Tested resync against an existing DCS parent that was not connected with Preservica, and this failed, as expected and intended

**Comment by motropuk** [comment]
Will test create new object with large (size) child next. Getting image from BRBL to add to Preservica TEST

**Comment by motropuk** [comment]
Add and delete child object rsync issue fixed by #2083. Still need to manually update the parent for the sync to complete. This should be fixed by #2105

**Comment by motropuk** [comment]
Testing reorder resync. Updated the filename for a child in Preservica Test that changed the order that that child appeared in the Preservica UI. Now running resync in DCS https://collections-uat.library.yale.edu/management/batch_processes/1113

**Comment by motropuk** [comment]
Created #2137

**Comment by motropuk** [comment]
Sort order resync was successful https://collections-uat.library.yale.edu/catalog/900145752. However, the batch process did not complete again https://collections-uat.library.yale.edu/management/batch_processes/1113

**Comment by motropuk** [comment]
I think we can close this but will need to keep looking into processes that dont complete as we do more testing of the Preservica integration before we roll it out
